Objects/object.c

```diff
@@ -2360,12 +2360,12 @@ new_reference(PyObject *op)
 #if !defined(Py_GIL_DISABLED)
     op->ob_refcnt = 1;
 #else
-    op->ob_tid = _Py_ThreadId();
+    _Py_atomic_store_uintptr_release(&op->ob_tid, _Py_ThreadId());
```

Comment:
I don't think we want to do this, at least not for now. I want to wait until we fix all the other races before we address `new_reference` because there are perf implications and the race is in practice benign.

The "release" memory order is also stronger (and more expensive on arm64) than necessary.